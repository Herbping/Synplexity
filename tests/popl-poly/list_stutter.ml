type 'a list =
  | Nil
  | Cons of 'a * 'a list

let rec list_stutter : 'a list -> 'a list |>
/\([] -> [],
   [a] -> [a; a],
   [b; a] -> [b; b; a; a]) = ?
