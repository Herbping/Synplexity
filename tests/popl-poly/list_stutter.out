(* Generated Thu Oct  8 20:27:08 2015 *)
fix f1 (l1:'a list) : 'a list ->
  match l1 with
  | Nil u1 -> Nil
  | Cons t1 -> Cons (#1 t1, Cons (#1 t1, f1 (#2 t1)))
